Stack.java

```diff
@@ -0,0 +1,141 @@
+package stack;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/*
+ * @author udit
+ */
+
+public class Stack<Type> implements Iterable<Type> {
+	private int size;
+	private SListNode<Type> head;
+
+	private class SListNode<Type2> {
+		public Type2 val;
+		public SListNode<Type2> next;
+		
+		public SListNode(Type2 i) {
+			this.val = i;
+			this.next = null;
+		}
+	}
+		
+	public Stack() {
```

Comment:
Explain your functions with appropriate comments

Reply:
It is necessary to add docStrings(documentation like multiline comments) at the top of your functions and methods. Please add them as per need